native/com_wolfssl_WolfSSL.c

```diff
@@ -1336,6 +1336,76 @@ JNIEXPORT jstring JNICALL Java_com_wolfssl_WolfSSL_getAvailableCipherSuitesIana
     return retString;
 }
 
+#ifdef WOLFSSLJNI_USE_NATIVE_CRYPTOCB
+/**
+ * Default native wolfSSL crypto callback function.
+ * This is called by default when wolfJSSE's WolfSSLProvider.registerDevId()
+ * is called, and is called by the native JNI API below.
+ *
+ * This function should be directly edited here to meet required
+ * functionality, or re-implemented and the registration point in the following
+ * function below should be edited:
+ *
+ * Java_com_wolfssl_WolfSSL_wc_1CryptoCb_1RegisterDevice()
+ */
+int DefaultNativeCryptoDevCb(int devId, wc_CryptoInfo* info, void* ctx)
+{
+    int ret = CRYPTOCB_UNAVAILABLE;
+    (void)devId;
+    (void)info;
+    (void)ctx;
+
+    /* Return CRYPTOCB_UNAVAILABLE to bypass HW and use SW. Edit function
+     * body here for your correct/expected functionality. */
+    return ret;
+}
+#endif /* WOLFSSLJNI_USE_NATIVE_CRYPTOCB */
+
+JNIEXPORT jint JNICALL Java_com_wolfssl_WolfSSL_wc_1CryptoCb_1RegisterDevice
+  (JNIEnv* jenv, jclass jcl, jint devId)
+{
+#ifdef WOLF_CRYPTO_CB
+    #ifdef WOLFSSLJNI_USE_NATIVE_CRYPTOCB
+        return wc_CryptoCb_RegisterDevice((int)devId,
+                    DefaultNativeCryptoDevCb, NULL);
+    #else /* Lookup the devId and see if it matches a known implementation */
+        #if defined(HAVE_CCBVAULTIC) && defined(WOLF_CRYPTO_CB_CMD)
+        #include "ccb_vaultic.h"
+        if(devId == CCBVAULTIC420_DEVID) {
+            return wc_CryptoCb_RegisterDevice((int)devId,
+                                              ccbVaultIc_CryptoDevCb, NULL);
+        }
+        #endif
+
+         /* could add additional elif blocks for ports / known callbacks */
+
+        /* No matching callback.  Maybe return CRYPTOCB_UNAVAILABLE? */
```

Comment:
I do think we should error out here if none are available or crypto callbacks are not compiled in. My thoughts are that if the user explicitly calls this function to register a device than silently continuing on could cause them to think they are using a registered callback when in fact they are not.

Reply:
Reasonable idea.  They will still get errors when they try to use the devid, so at least it won't be invisible.  Let's revisit this when we figure out how to do this for reals.  I'm thinking we should provide some simple java stubs so register(int) would really be register(int, cb, config).

Reply:
Good feedback, adjusted this to return `CRYPTOCB_UNAVAILABLE` if no callback has been called.